core/web/api/observable.py

```diff
@@ -46,13 +46,16 @@ def new(self):
         params = request.json
 
         if 'id' in params:
-            obs = self.objectmanager.objects.get(id=params.pop("id"))
+            obs = get_object_or_404(self.objectmanager, id=params.pop("id", None))
         else:
             if params.pop('refang', None):
-                obs = self.objectmanager.add_text(refang(params.pop('value')))
+                obs = self.objectmanager.add_text(refang(params.pop('value')), force_type=params.pop('force_type', None))
```

Comment:
I would extract the `params.pop('force_type', None)` to a variable right before the condition on `refang`

e.g.

```python
forced_type = params.pop('force_type', None)
if params.pop('refang', None):
    obs = self.objectmanager.add_text(refang(params.pop('value')), force_type=forced_type)
else:
    obs = self.objectmanager.add_text(params.pop('value'), force_type=forced_type)
```